src/rotate.jl

```diff
@@ -0,0 +1,102 @@
+
+#
+# definition for rotating points
+# 
+
+"""
+Rotate the point X using the rotation parameters specified in R (i.e. R * X)
+
+rotate_point(R::T, X)
+
+"""
+rotate_point{T <: RotationTypes, U <: Real}(R::T, X::Vec{3,U})  =  rotate_point(RotMatrix(R), X)    # generic version, go through a rotation matrix
+
+# in case of mutable input (not the recomended form)
+rotate_point{T <: RotationTypes, U <: Real}(R::T, X::Vector{U}) = rotate_point(R, Vec{3,U}(X[1], X[2], X[3]))   # generic version, go through a rotation matrix
+
+# cases where we start with a rotation matrix with matched element types
+rotate_point{T <: Real}(R::RotMatrix{T}, X::Vec{3,T}) =   R * X
```

Comment:
Again, I would be happy if all of these accepted any `FixedVector{3}` and any `Vector`.